--- a/include/linux/ar8216_platform.h
+++ b/include/linux/ar8216_platform.h
@@ -44,6 +44,7 @@ struct ar8327_pad_cfg {
 	bool pipe_rxclk_sel;
 	bool txclk_delay_en;
 	bool rxclk_delay_en;
+	bool sgmii_delay_en;
 	enum ar8327_clk_delay_sel txclk_delay_sel;
 	enum ar8327_clk_delay_sel rxclk_delay_sel;
 };
@@ -75,4 +75,6 @@ struct ar8327_platform_data {
 	struct ar8327_pad_cfg *pad5_cfg;
 	struct ar8327_pad_cfg *pad6_cfg;
	struct ar8327_port_cfg cpuport_cfg;
+	struct ar8327_port_cfg port0_cfg;
+	struct ar8327_port_cfg port6_cfg;
 	struct ar8327_led_cfg *led_cfg;
 };
 
-#endif /* AR8216_PLATFORM_H */
\ No newline at end of file
+#endif /* AR8216_PLATFORM_H */
--- a/include/linux/phy.h
+++ b/include/linux/phy.h
@@ -417,6 +417,12 @@ struct phy_driver {
 	 */
 	int (*did_interrupt)(struct phy_device *phydev);
 
+	/*
+	 * Called before an ethernet device is detached
+	 * from the PHY.
+	 */
+	void (*detach)(struct phy_device *phydev);
+
 	/* Clears up any memory if needed */
 	void (*remove)(struct phy_device *phydev);
 
